### PR TITLE
Preserve property descriptors

### PR DESCRIPTION
**Pull request by papb**
This PR:

* Fixes tests that weren't really testing the order
* Updates the sorting mechanism to keep property descriptors intact

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -1,17 +1,49 @@
...
 test('sort the keys of an object', t => {
-	t.deepEqual(sortKeys({c: 0, a: 0, b: 0}), {a: 0, b: 0, c: 0});
+	deepEqualInOrder(t, sortKeys({c: 0, a: 0, b: 0}), {a: 0, b: 0, c: 0});
 });
 
 test('custom compare function', t => {
-	const compare = (a, b) => a.localeCompare(b);
-	t.deepEqual(sortKeys({c: 0, a: 0, b: 0}, {compare}), {c: 0, b: 0, a: 0});
+	const compare = (a, b) => b.localeCompare(a);
+	deepEqualInOrder(t, sortKeys({c: 0, a: 0, b: 0}, {compare}), {c: 0, b: 0, a: 0});
```
Comment (papb) [inline test.js:42]: After introducing `deepEqualInOrder`, it turned out this test was actually incorrect.

File: test.js [hunk 2]
```diff
@@ -1,17 +1,49 @@
...
+
+		if (seen.has(object1) && seen.has(object2)) {
+			return;
+		}
+
+		seen.add(object1);
+		seen.add(object2);
+
+		if (Array.isArray(object1)) {
+			for (let i = 0; i < object1.length; i++) {
```
Comment (sindresorhus) [inline test.js:20]: You can use `Object.entries` here and a for-of loop.

File: test.js (hunk 1 above)
Comment (sindresorhus) [reply test.js:42]: Nice catch ❤️

File: test.js (hunk 2 above)
Comment (papb) [reply test.js:20]: Hmm, indeed I can, but personally I think that would be less readable here. In particular, [`Object.entries` is documented to return **string-keyed** props of **objects**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries), and the fact that it works on arrays is probably an obscure consequence of the early days of JavaScript.

```js
Object.entries([10, 20]);
//=> [['0', 10], ['1', 20]]
```

And it returns indexes as strings, which I don't like.

Can I leave it this way? If not, I can

Comment (papb) [reply test.js:20]: Hmm, I guess you intended to say `Array.prototype.entries`. That would make more sense.

I still prefer the way I wrote it, but I don't mind switching in this case.

Comment (papb) [reply test.js]: I ended up using `Array.prototype.keys` instead. What do you think? 13ae015

Comment (sindresorhus) [reply test.js]: Yeah, looks good. I'm just staying away from normal loops as they often have the chance of off-by-one errors.

**Pull request merged**